### PR TITLE
OSSM-5690: Sync with istio 1.20

### PR DESCRIPTION
**Pull request by yannuil**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,72 +1,75 @@
...
 	github.com/spf13/pflag v1.0.5
-	istio.io/client-go v1.18.6
-	k8s.io/api v0.27.0
-	k8s.io/apimachinery v0.27.0
-	k8s.io/client-go v0.27.0
-	k8s.io/code-generator v0.27.0
-	k8s.io/gengo v0.0.0-20220902162205-c0856e24416d
-	k8s.io/klog/v2 v2.90.1
-	sigs.k8s.io/gateway-api v0.6.2
+	istio.io/client-go v1.20.0-rc.0
```
Comment (jewertow) [inline go.mod:14]: Shouldn't it be `istio.io/client-go v1.20.2-0.20231213021015-d9f0411d8b67`, like in [maistra/istio](https://github.com/maistra/istio/pull/920/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R113)?

Comment (yannuil) [reply go.mod:14]: I want to make it the same as the istio [commit](https://github.com/istio/istio/commit/6869a6c2371e21c847d216065cf5c59863d01b4c) 1.20 I am rebasing on, I can eliminate the extra changes in the go.mod commit.

**Review by jwendell — APPROVED** [review]

**Pull request merged**